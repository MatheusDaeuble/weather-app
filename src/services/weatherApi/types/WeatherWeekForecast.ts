import { IconName } from 'ui/components/atoms/Icon/types';
import { IllustrationName } from 'ui/components/atoms/Illustration/types';

export default interface WeatherWeekForecast {
  timestamp: number;
  min: number;
  max: number;
  icon: IconName;
  illustration: IllustrationName;
}
