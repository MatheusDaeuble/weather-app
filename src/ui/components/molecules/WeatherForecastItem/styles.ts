import styled from 'styled-components/native';
import Typography from 'ui/components/atoms/Typography';
import { Spacing } from 'ui/theme';

export const Container = styled.View`
  width: 100%;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-vertical: ${Spacing.SCALE_8}px;
`;

export const Temp = styled(Typography).attrs({
  variant: 'h5',
  color: 'gray',
  align: 'left',
})``;

export const Day = styled(Typography).attrs({
  color: 'gray',
})`
  flex: 1;
`;

export const Empty = styled(Typography).attrs({
  color: 'gray',
  children: 'Não foi possível carregar as previsões.',
  align: 'center',
})`
  flex: 1;
`;

export const TempContainer = styled.View`
  flex: 1;
  align-items: center;
  flex-direction: row;
  justify-content: flex-end;
`;
