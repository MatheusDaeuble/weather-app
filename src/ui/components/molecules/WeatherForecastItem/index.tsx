import React from 'react';
import { isSameDay } from 'date-fns';
import { formatTemperature, formatWeekDay } from 'util/format';
import Icon from 'ui/components/atoms/Icon';
import { Container, Day, Temp, TempContainer } from './styles';
import { WeatherWeekForecast } from 'services/weatherApi/hooks/useWeatherApi copy';

interface WeatherForecastItemProps {
  data: WeatherWeekForecast;
}

const WeatherForecastItem = ({ data }: WeatherForecastItemProps) => {
  const isToday = isSameDay(data.timestamp, new Date().getTime());

  return (
    <Container>
      <Day variant={isToday ? 'h4' : 'h5'}>
        {isToday ? 'Hoje' : formatWeekDay(data.timestamp)}
      </Day>
      <Icon name={data.icon} size={18} />
      <TempContainer>
        <Icon name='chevron-up' color='warning' size={18} />
        <Temp>{formatTemperature(data.max)} </Temp>
        <Icon name='chevron-down' color='secondary' size={18} />
        <Temp>{formatTemperature(data.min)} </Temp>
      </TempContainer>
    </Container>
  );
};
export default WeatherForecastItem;
