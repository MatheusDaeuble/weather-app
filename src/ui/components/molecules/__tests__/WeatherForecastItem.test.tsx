import React from 'react';
import renderer from 'react-test-renderer';
import WeatherWeekForecast from 'services/weatherApi/types/WeatherWeekForecast';
import WeatherForecastItem from '../WeatherForecastItem';

const defaultItem: WeatherWeekForecast = {
  timestamp: Date.now(),
  min: 16.55,
  max: 22.33,
  icon: 'cloudy',
  illustration: 'cloud',
};

describe('WeatherForecastItem Molecule', () => {
  it('Today', () => {
    const component = renderer
      .create(<WeatherForecastItem data={defaultItem} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Not today', () => {
    const component = renderer
      .create(
        <WeatherForecastItem
          data={{
            ...defaultItem,
            timestamp: new Date('2022-06-05').getTime(),
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
