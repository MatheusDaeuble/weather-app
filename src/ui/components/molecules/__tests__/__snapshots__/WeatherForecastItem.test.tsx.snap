// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForecastItem Molecule Not today 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 16,
        "width": "100%",
      },
    ]
  }
>
  <Text
    align="left"
    color="gray"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#949494",
          "fontFamily": "OpenSans_400Regular",
          "fontSize": 32,
          "lineHeight": 48,
          "textAlign": "left",
        },
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    variant="h5"
  >
    Sábado
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "width": 36,
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": 36,
        }
      }
    >
      <Text />
    </View>
    <Text
      align="left"
      color="gray"
      paragraph={false}
      style={
        Array [
          Object {
            "color": "#949494",
            "fontFamily": "OpenSans_400Regular",
            "fontSize": 32,
            "lineHeight": 48,
            "textAlign": "left",
          },
          Object {},
        ]
      }
      variant="h5"
    >
      22°
       
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": 36,
        }
      }
    >
      <Text />
    </View>
    <Text
      align="left"
      color="gray"
      paragraph={false}
      style={
        Array [
          Object {
            "color": "#949494",
            "fontFamily": "OpenSans_400Regular",
            "fontSize": 32,
            "lineHeight": 48,
            "textAlign": "left",
          },
          Object {},
        ]
      }
      variant="h5"
    >
      17°
       
    </Text>
  </View>
</View>
`;

exports[`WeatherForecastItem Molecule Today 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 16,
        "width": "100%",
      },
    ]
  }
>
  <Text
    align="left"
    color="gray"
    paragraph={false}
    style={
      Array [
        Object {
          "color": "#949494",
          "fontFamily": "OpenSans_700Bold",
          "fontSize": 32,
          "lineHeight": 48,
          "textAlign": "left",
        },
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    variant="h4"
  >
    Hoje
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "width": 36,
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": 36,
        }
      }
    >
      <Text />
    </View>
    <Text
      align="left"
      color="gray"
      paragraph={false}
      style={
        Array [
          Object {
            "color": "#949494",
            "fontFamily": "OpenSans_400Regular",
            "fontSize": 32,
            "lineHeight": 48,
            "textAlign": "left",
          },
          Object {},
        ]
      }
      variant="h5"
    >
      22°
       
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": 36,
        }
      }
    >
      <Text />
    </View>
    <Text
      align="left"
      color="gray"
      paragraph={false}
      style={
        Array [
          Object {
            "color": "#949494",
            "fontFamily": "OpenSans_400Regular",
            "fontSize": 32,
            "lineHeight": 48,
            "textAlign": "left",
          },
          Object {},
        ]
      }
      variant="h5"
    >
      17°
       
    </Text>
  </View>
</View>
`;
